### ISSUE TITLE
Add Google Cloud Storage log format

### ISSUE THREAD

**Issue by nikmartin** [allinurl/goaccess#262]
The format is here:

https://cloud.google.com/storage/docs/access-logs#format

I'm unable to create a custom format, but I don't know how to debug. My suspicion lies in the fact that the files are CSV, and the time format is in microseconds

**Comment by nikmartin** [comment]
For reference, here is the top few lines of an actual log file:

```
"time_micros","c_ip","c_ip_type","c_ip_region","cs_method","cs_uri","sc_status","cs_bytes","sc_bytes","time_taken_micros","cs_host","cs_referer","cs_user_agent","s_request_id","cs_operation","cs_bucket","cs_object"
"1435673425692000","180.76.15.137","1","","GET","/blog/data-ownership-in-nitropcr","301","0","0","211000","www.nitropcr.com","","Mozilla/5.0 (compatible; Baiduspider/2.0; +http://www.baidu.com/search/spider.html),gzip(gfe)","AEnB2Uo6_oFt6zlbUfcwodxyyKbvVNHi-PytOKYHJuxBOdWZgwcUe-ZGcBhJaCQwi3xS0wSaK7OPoVXmwYF2pwWH8L4s16TzoA","GET_Object","www.nitropcr.com","blog/data-ownership-in-nitropcr"
"1435673426067000","180.76.15.139","1","","GET","/blog/data-ownership-in-nitropcr/index.html","200","0","13654","179000","www.nitropcr.com","","Mozilla/5.0 (compatible; Baiduspider/2.0; +http://www.baidu.com/search/spider.html),gzip(gfe)","AEnB2UqRS8KJ9c9WfLWhpK-lG2wPir9rII5Dt0Dckw6g6oj-BVLXCMsqBpZQZ-11g4hRKp0mGQp4csi3NrQwGPXig3BPwLN7hQ","GET_Object","www.nitropcr.com","blog/data-ownership-in-nitropcr/index.html"
"1435673418492001","2602:30a:c037:b3b0:d63d:7eff:fef1:49b5","2","","GET","/storage/v1/b/www.nitropcr.com?fields=logging&alt=json&projection=full","200","0","97","50000","www.googleapis.com","","apitools gsutil/4.13 (linux2) Cloud SDK Command Line Tool 0.9.65,gzip(gfe)","b2ae6a88c12b69f55b5d3397cca80ba2","storage.buckets.get","www.nitropcr.com",""
"1435673978352000","66.249.79.253","1","","GET","/robots.txt","200","0","24","261000","www.nitropcr.com","","Mozilla/5.0 (compatible; Googlebot/2.1; +http://www.google.com/bot.html),gzip(gfe)","AEnB2Ur1ACazfDlNV-J3qw6_9bKb6YxmQQao7ZidoFgzl0A0RZTOe_Okoq0UibSYN2MwxvWG2rRPUc1ov0E1S8reNK0Eq49i_Q","GET_Object","www.nitropcr.com","robots.txt"
```
The file always has the header in it

**Comment by allinurl** [comment]
`strptime` does not support microseconds (w/o extensions). I can add a field descriptor for this.

As far as the `log-format`, the following would work (tested with `%s` for date/time-format):

    "%d","%h",%^,%^,"%m","%U","%s",%^,"%b","%D",%^,"%R","%u"

**Comment by nikmartin** [comment]
Hmm, let me try again, I think that's what I tried last with no success

**Comment by nikmartin** [comment]
Yes, that did work, I was trying to do this with the time and date format: `%s%6` based on strftime

**Comment by allinurl** [comment]
The ability to parse Google Cloud Storage access logs has been added. I also added a predefined format to the config dialog list. The formats to use are:

    log-format "%x","%h",%^,%^,"%m","%U","%s",%^,"%b","%D",%^,"%R","%u"
    date-format  %f
    time-format  %f

This will be pushed out in the upcoming version, otherwise, feel free to build from [master](http://goaccess.io/download#build).
